File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.cpp
class Solution {
public:
    int findGCD(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(),nums.end());
        int small = nums[0];
        int large = nums[n-1];
        
        for(int i=small;i>=0;i--){
            if(small % i == 0 && large % i == 0){
                return i;
            }
        }

        return 1;
    }
};

File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {
        int small = nums[0];
        int large = nums[0];

        
        for (int num : nums) {
            if (num < small) small = num;
            if (num > large) large = num;
        }

        
        for (int i = small; i > 0; i--) {
            if (small % i == 0 && large % i == 0) {
                return i;
            }
        }

        return 1;
    }
}